homeassistant/components/ads/binary_sensor.py

```diff
@@ -1,12 +1,15 @@
 """Support for ADS binary sensors."""
 import logging
+import asyncio
+import async_timeout
 
 import voluptuous as vol
 
 from homeassistant.components.binary_sensor import (
     DEVICE_CLASSES_SCHEMA, PLATFORM_SCHEMA, BinarySensorDevice)
 from homeassistant.const import CONF_DEVICE_CLASS, CONF_NAME
 import homeassistant.helpers.config_validation as cv
+from homeassistant.util.async_ import run_coroutine_threadsafe
```

Comment:
I suggest we use the builtin asyncio function:
https://docs.python.org/3/library/asyncio-task.html#asyncio.run_coroutine_threadsafe

This asyncio util function is legacy from when we ran Python 3.4.

Reply:
done